test/ScaledDomains.Extensions.Caching.MySql.Tests/MySqlServerCacheMaintenanceTests.cs

```diff
@@ -14,65 +13,75 @@ namespace ScaledDomains.Extensions.Caching.MySql.Tests
     [TestCategory(TestCategoryNames.Integration)]
     public class MySqlServerCacheMaintenanceServiceTests : IntegrationTestBase
     {
-
         [TestMethod]
         public void CreateAnInstance_WithValidConfiguration_ShouldCreate()
         {
-            _ = new MySqlServerCacheMaintenanceService(new MySqlServerCacheOptions
-            {
-                ConnectionString = "Server=example.com;Database=db;User=root;",
-                TableName = "table"
-            });
+            _optionsMock.SetupGet(o => o.Value).Returns(() => new MySqlServerCacheOptions());
+            _ = CreateMySqlServerCacheMaintenanceService(_optionsMock.Object, _loggerMock.Object, _databaseOperationsMock.Object);
+        }
+
+        private static MySqlServerCacheMaintenanceService CreateMySqlServerCacheMaintenanceService(IOptions<MySqlServerCacheOptions> options, ILogger<MySqlServerCacheMaintenanceService> logger, IDatabaseOperations databaseOperations)
+        {
+            return new MySqlServerCacheMaintenanceService(options, logger, databaseOperations);
         }
 
         [TestMethod]
         [ExpectedException(typeof(ArgumentNullException))]
         public void CreateAnInstance_WithNullConfiguration_ShouldThrowArgumentNullException()
         {
-            _ = new MySqlServerCacheMaintenanceService(null);
+            _ = CreateMySqlServerCacheMaintenanceService(null!, _loggerMock.Object, _databaseOperationsMock.Object);
         }
 
         [TestMethod]
-        public async Task ExecuteAsync_DeleteExpiredCacheItemsAsyncInvokeOnce()
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void CreateAnInstance_WithNullLogger_ShouldThrowArgumentNullException()
         {
-            // Arrange
+            _optionsMock.SetupGet(o => o.Value).Returns(() => new MySqlServerCacheOptions());
+            _ = CreateMySqlServerCacheMaintenanceService(_optionsMock.Object, null!, _databaseOperationsMock.Object);
+        }
 
-            var databaseOperationMock = new Mock<IDatabaseOperations>();
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void CreateAnInstance_WithNullDatabaseOperations_ShouldThrowArgumentNullException()
+        {
+            _optionsMock.SetupGet(o => o.Value).Returns(() => new MySqlServerCacheOptions());
+            _ = CreateMySqlServerCacheMaintenanceService(_optionsMock.Object, _loggerMock.Object, null!);
+        }
 
-            var config = TestConfiguration.MySqlServerCacheOptions.Clone();
-            config.DatabaseOperations = databaseOperationMock.Object;
+        [TestMethod]
+        public async Task ExecuteAsync_DeleteExpiredCacheItemsAsyncInvokeOnce()
```

Comment:
If you remove the invoke count check you should update name of the test method to this: 
`ExecuteAsync_DeleteExpiredCacheItemsAsyncInvoked`

Reply:
Added back the Times check.